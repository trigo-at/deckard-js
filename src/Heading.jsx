import React from 'react';
import {oneOf} from 'prop-types';
import cx from 'classnames';

const Heading = ({size, as, ...props}) => {
    const Component = as;
    return (
        <Component
            className={cx('tracking-wide font-normal leading-loose', {
                'text-4xl': size === 'xl',
                'text-2xl': size === 'l',
                'text-xl': size === 'm',
                'text-lg': size === 's',
            })}
            {...props}
        />
    );
};

Heading.propTypes = {
    as: oneOf(['h1', 'h2', 'h3', 'h4']),
    size: oneOf(['xl', 'l', 'm', 's']),
};

Heading.defaultProps = {
    as: 'h1',
    size: 'xl',
};

export default Heading;
