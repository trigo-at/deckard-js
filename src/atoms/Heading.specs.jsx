import React from 'react';
import renderer from 'react-test-renderer';
import Heading from './Heading';

test('Heading matches snapshot', () => {
    const component = renderer.create(<Heading>trigo</Heading>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
