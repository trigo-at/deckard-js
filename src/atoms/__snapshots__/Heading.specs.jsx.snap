// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading matches snapshot 1`] = `
.c0 {
  font-size: 24px;
  margin: 0px;
}

<h3
  className="c0"
  fontSize={4}
>
  trigo
</h3>
`;
